src/components/CommentAdder.jsx

```diff
@@ -0,0 +1,80 @@
+import { useContext, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { UserContext } from '../contexts/UserContext';
+import { postComment } from '../utils';
+import { Link } from 'react-router-dom';
+
+export default function CommentAdder({ setComments }) {
+  const [newComment, setNewComment] = useState('');
+  const { user } = useContext(UserContext);
+  const { article_id } = useParams();
+  const [loginAlert, setLoginAlert] = useState(false);
+  const [commentAlert, setCommentAlert] = useState(false);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    if (user.username === 'Logged Out') {
+      setLoginAlert(true);
+      return;
+    }
+
+    if (newComment.length === 0) {
+      setCommentAlert(true);
+      return;
+    }
+
+    postComment(user.username, newComment, article_id)
+      .then((newCommentFromApi) => {
+        setNewComment('');
+        setComments((currComments) => {
+          return [...currComments, newCommentFromApi];
+        });
+      })
+      .catch((err) => {
+        console.log(
+          `Failed to add comment for ${article_id}. Please try again.`
+        );
+      });
+  };
+
+  const handleTextareaFocus = () => {
+    setCommentAlert(false);
+  };
+
+  return (
```

Comment:
Really great ammount of features in this so far Tom, great work.

Couple things to add are:
1. Preventing duplicate clicks of the post comment button from posting the same comment twice.
2. Some feedback for succesfull posts.